src/averageAges.js

```diff
@@ -15,11 +15,22 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+  const ageOfDeath = men.map(person => person.died - person.born);
+  const centureResultArr = men.filter(person =>
+    Math.ceil(person.died / 100) === century
+  );
+  const diedThisCentury = centureResultArr.map(pers =>
+    pers.died - pers.born
+  );
+
+  const result = !century ? (
+    ageOfDeath.reduce((sum, age) => sum + age, 0) / men.length
+  ) : (
+    diedThisCentury.reduce((sum, age) => sum + age, 0) / centureResultArr.length
+  );
```

Comment:
when you do like I said previously you won't need condition here, you'll need only one reduce